sys/dev/cxgbe/tom/t4_cpl_io.c

```diff
@@ -1967,16 +1967,16 @@ alloc_aiotx_mbuf(struct kaiocb *job, int len)
 	 */
 	vm = job->userproc->p_vmspace;
 	map = &vm->vm_map;
-	start = (uintptr_t)job->uaiocb.aio_buf + job->aio_sent;
-	pgoff = start & PAGE_MASK;
+	start = __DEVOLATILE(char * __capability, job->uaiocb.aio_buf) + job->aio_sent;
+	pgoff = (__cheri_addr vm_offset_t)start & PAGE_MASK;
```

Comment:
This pattern crops up a bit; I wonder if we want a page_offset macro or something like we have {round,trunc}_page.

Reply:
Indeed, a dumb grep from `& PAGE_MASK;` turns up over 100 of them.

Reply:
I'm interested in making this change, but would like to get upstream buyin before I make a diff this large.